File: Python/Rotate.py
# Assume you have a method isSubstringwhich checks if oneword is a substring of another. 
# Given two strings, sl and s2, write code to check if s2 is a rotation of sl 
#using only one call to isSubstring 
class Rotate:
    def is_rotation(self, str1, str2):
        if str1 is None or str2 is None or len(str1) != len(str2):
            return False
        
        if str1 == "" and str2 == "":
            return True
        
        str11 = str1 + str1
        return self.is_substring(str11, str2)
    
    def is_substring(self, str11, str2):
        length = len(str2)
        for i in range(len(str11)):
            print(str11[i:i+length])
            if str11[i:i+length] == str2:
                return True
        return False

if __name__ == "__main__":
    r = Rotate()
    c1 = input("Ingrese la primera palabra: ")
    c2 = input("Ingrese la segunda palabra: ")
    print(r.is_rotation(c1, c2))

File: Java/Rotate.java
import java.util.Scanner;
/* Assume you have a method isSubstringwhich checks if oneword is a substring of another. 
Given two strings, sl and s2, write code to check if s2 is a rotation of sl 
using only one call to isSubstring  */

public class Rotate {
    public boolean isRotation(String str1, String str2){
        if (str1 == null || str2 == null || str1.length() != str2.length()) 
            return false;

        if(str1.equals("") && str2.equals(""))
            return true;

        String str11 = str1+str1;
        return isSubstring(str11, str2);    
    }

    private boolean isSubstring(String str11, String str2){
        int len = str2.length(); 
        for(int i = 0; i < len; i++){
            System.out.println(str11.subSequence(i, i+len));
			if(str11.subSequence(i, i+len).equals(str2))
				return true;			
        }
        return false;
    }

    public static void main(String[] args) {
        Scanner sn = new Scanner(System.in);
        Rotate r = new Rotate();
        System.out.print("Ingrese dos palabras: ");
        String c1 = sn.nextLine();
        String c2 = sn.nextLine();
        System.out.println(r.isRotation(c1, c2));
    }
}